lib/cloudkeeper/managers/image_manager.rb

```diff
@@ -63,10 +63,13 @@ def retrieve_image(uri, filename)
           Net::HTTP.start(uri.host, uri.port) do |http|
             request = Net::HTTP::Get.new(uri)
 
-            http.request(request) { |response| open(filename, 'w') { |file| response.read_body { |chunk| file.write(chunk) } } }
+            http.request(request) do |response|
+              response.value
```

Comment:
Apparently that's how one uses Ruby's HTTP library. `value` method called on response will do nothing on any 2xx status code and will throw an exception otherwise.